### PR TITLE
Arc 1539 audience tech planning create parameterized macros

### PR DESCRIPTION
**Pull request by ryantimjohn**
# Pull Request for dbt-arc-functions

- [x] Open the pull request as a draft so that you can run tests against the code and inspect the files in the Files view of the Pull Request interface in Github.

- [x] Once you're satisfied, convert the draft into an open Pull Request by clicking "Ready for Review" near the bottom of this page. You should not make any changes to the code except in dire emergency once you've converted from draft, so make sure you're really satisfied.

- [x] Add two of the principal contributors to this PR for review. Currently, the principal contributors are @dmbluestate , @Frydafly , and @ryantimjohn.

- [x] We've divided our PR templates into Macros, Utils and Docs PR requests. Delete the ones you're not using.
### Macros (delete if not using)

- [x] Link Jira tickets (or Github Issues) to PR here. Make sure that the Jira ticket has a good description as to why this PR is necessary and the business problem that this PR is solving. Moreover, if there have been any discussions of this ticket in Slack, make sure they're linked in the Jira ticket:
https://bluestate.atlassian.net/browse/ARC-1539

- [x] Make sure this is named using Jira extension so that naming convention is consistent. If it's not, that's okay. Go to the Jira ticket, click into the ticket, then on the right under Details, find Development, then click the Create branch button. Copy the Branch Name from there. Read here [how to rename a branch in Github](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) to rename this branch. In the future, create branches from Jira.

- [x] Make sure the Macro is documented by running the notebook: `create_docs.ipynb` util in this repo. From there, you'll have to either fill in descriptions by hand OR use our `fill_in_descriptions_using_openai.ipynb` to fill in descriptions using OpenAI's ChatGPT. Always make sure to check over ChatGPT's column descriptions, as they're likely not perfect.

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below.
Tested!

- [x] If new macro files were created or updated, re-run `create_or_update_standard_models.py` for the client dbt project, replacing models and dependencies

Ran several times successfully, including when updating the model!

Screenshot 2023-07-20 at 12 16 45 PM

- [x] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page

Screenshot 2023-07-20 at 12 02 41 PM

Here is what the new model code looks like:
```sql
-- ***WARNING The following parameters must be filled for this file to work:***
-- ['best_guess_inbound_channel']
-- macro used to create this file can be found at:
-- https://github.com/bsd/dbt-arc-functions/blob/main/macros/stitch_sfmc_audience_transaction/staging/stg_stitch_sfmc_parameterized_audience_transactions_enriched.sql
{{ dbt_arc_functions.create_stg_stitch_sfmc_audience_transactions_enriched(    
    best_guess_inbound_channel="case
        when (lower(source_code) like 'email_%' or lower(source_code) like '%bsd%') then 'Email'
        when lower(source_code) like 'web%' then 'Web'
        when (lower(source_code) like 'cpc_%' or lower(source_code) like 'sem_%') then 'CPC'
        when (lower(source_code) like 'ref_%' or lower(source_code) like 'referral_%') then 'Referral'
        when lower(source_code) like 'organic_%' then 'Organic'
        when lower(source_code) like 'paidsocial_%' then 'Paid Social'
        when lower(source_code) like 'display%' then 'Display'
        when lower(source_code) like 'social%' then 'Social'
        when source_code is null then 'No Source'
        else inbound_channel
      end",
    reference_name="stg_stitch_sfmc_audience_transactions_summary_unioned"
) }}
```

And here's what it looks like at compile time:
```sql
-- ***WARNING The following parameters must be filled for this file to work:***
-- ['best_guess_inbound_channel']
-- macro used to create this file can be found at:
-- https://github.com/bsd/dbt-arc-functions/blob/main/macros/stitch_sfmc_audience_transaction/staging/stg_stitch_sfmc_parameterized_audience_transactions_enriched.sql


    select
        transaction_id,
        person_id,
        transaction_date_day,
        amount,
        inbound_channel,
        (case
        when (lower(source_code) like 'email_%' or lower(source_code) like '%bsd%') then 'Email'
        when lower(source_code) like 'web%' then 'Web'
        when (lower(source_code) like 'cpc_%' or lower(source_code) like 'sem_%') then 'CPC'
        when (lower(source_code) like 'ref_%' or lower(source_code) like 'referral_%') then 'Referral'
        when lower(source_code) like 'organic_%' then 'Organic'
        when lower(source_code) like 'paidsocial_%' then 'Paid Social'
        when lower(source_code) like 'display%' then 'Display'
        when lower(source_code) like 'social%' then 'Social'
        when source_code is null then 'No Source'
        else inbound_channel
      end) as best_guess_inbound_channel,
        recurring,
        (
            case
                when amount between 0 and 25
                then '0-25'
                when amount between 26 and 100
                then '26-100'
                when amount between 101 and 250
                then '101-250'
                when amount between 251 and 500
                then '251-500'
                when amount between 501 and 1000
                then '501-1000'
                when amount between 1001 and 10000
                then '1001-10000'
                else '10000+'
            end
        ) as gift_size_string,
        row_number() over (
            partition by person_id order by transaction_date_day
        ) as gift_count
    from `bsd-arc-uusa`.`dbt_ryantimjohn_staging`.`stg_stitch_sfmc_audience_transactions_summary_unioned`
```
You can see what code changes look like here:
https://github.com/bsd/arc-uusa-dbt/pull/13

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this

### Utils (delete if not using)

- [x] Suggest any tests that your PR reviewer might use to check the utils here. Better yet, if you're able, write automated tests in Github Actions!
Run `python create_or_update_standard_models.py` and try and update the `stitch_sfmc_audience_transaction` standard models!

- [x] Update utils/README.md with any new utils. Make sure that the Table of Contents is updated (most IDEs/code editors can do this for you!)

### CODE REVIEW

File: macros/stitch_sfmc_audience_transaction/staging/stg_stitch_sfmc_parameterized_audience_transactions_enriched.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{% macro create_stg_stitch_sfmc_audience_transactions_enriched(
```
Comment (Frydafly) [inline macros/stitch_sfmc_audience_transaction/staging/stg_stitch_sfmc_parameterized_audience_transactions_enriched.sql:1]: macro name doesn't reflect the model? doesn't have parameterized

Comment (ryantimjohn) [reply macros/stitch_sfmc_audience_transaction/staging/stg_stitch_sfmc_parameterized_audience_transactions_enriched.sql:1]: ty!

**Comment by ryantimjohn** [comment]
Just so shenanigans like what @Frydafly pointed out above don't happen again I'm adding a ticket to make a linter check that macro names match their filenames:
https://bluestate.atlassian.net/browse/ARC-1552

**Comment by ryantimjohn** [comment]
Screenshot 2023-07-21 at 1 37 47 PM
Reran

**Review by dmbluestate — APPROVED** [review]
Tested and runs properly. Will def still need to revisit this approach later as the use case will need to be adjusted, as reflected in https://bluestate.atlassian.net/browse/ARC-1558

**Pull request merged**